src/client/eppo-client.ts

```diff
@@ -1199,6 +1209,78 @@ export default class EppoClient {
       sdkLibVersion: LIB_VERSION,
     };
   }
+
+  private getAllBandits(
+    subjectKey: string,
+    subjectAttributes: ContextAttributes,
+    banditActions: Record<string, BanditActions>,
+    flags: Record<string, PrecomputedFlag>,
+  ): Record<string, IPrecomputedBandit> {
+    const banditResults: Record<string, IPrecomputedBandit> = {};
+
+    Object.keys(banditActions).forEach((flagKey: string) => {
+      // First, check how the flag evaluated.
+      const flagVariation = flags[flagKey];
+      if (flagVariation) {
+        // Precompute a bandit, if there is one matching this variation.
+        const precomputedResult = this.getPrecomputedBandit(
+          flagKey,
+          flagVariation.variationValue,
+          subjectKey,
+          subjectAttributes,
+          banditActions[flagKey],
+        );
+        if (precomputedResult) {
+          banditResults[flagKey] = precomputedResult;
+        }
+      }
+    });
+    return banditResults;
+  }
+
+  private findBanditByVariation(flagKey: string, variationValue: string): BanditParameters | null {
+    const banditVariations = this.banditVariationConfigurationStore?.get(flagKey);
+    const banditKey = banditVariations?.find(
+      (banditVariation) => banditVariation.variationValue === variationValue,
+    )?.key;
+
+    if (banditKey) {
+      // Retrieve the model parameters for the bandit
+      return this.getBandit(banditKey);
+    }
+    return null;
+  }
+
+  private getPrecomputedBandit(
+    flagKey: string,
+    variationValue: string,
+    subjectKey: string,
+    subjectAttributes: ContextAttributes,
+    banditActions: BanditActions,
+  ): IPrecomputedBandit | null {
+    const bandit = this.findBanditByVariation(flagKey, variationValue);
+    if (bandit) {
```

Comment:
same style suggestion here, handle undefined bandit and flatten code below

Reply:
👍